src/components/Thumbnail.vue

```diff
@@ -0,0 +1,33 @@
+<template>
+    <div class="canvas-container" ref="canvasContainer"></div>
+</template>
+
+<script setup lang="ts">
+    import {onMounted, ref} from 'vue'
+    import {Specimen} from '@/shared/Specimen'
+
+    const canvasContainer = ref<HTMLDivElement | null>(null)
+    const props = defineProps(['base64'])
+
+    onMounted(() => {
+        const specimen = Specimen.decode64(props.base64)
+
+        if (!(canvasContainer.value instanceof HTMLElement)) return
+
+        specimen.visualizer.validate()
+        specimen.sequence.validate()
+        specimen.visualizer.isValid = true
+        specimen.sequence.isValid = true
+        specimen.visualizer.assignParameters()
+        specimen.sequence.assignParameters()
```

Comment:
The fact that we have to call validate, assign isValid, and call assignParameters on each of the sequence and visualizer before we are ready to setup() is a pretty clear indication there is a poor factoring somewhere. I would say at least at the end of a successful validate() the elements should have isValid() true. And it seems quite possible to me that the decoding that created the specimen should have assigned the parameters as well. I think we should track this down and get the Paramable api here rationalized before we merge this.

Reply:
this came up again in getting the name of a sequence, so definitely needs to be addressed.

Reply:
Such repeated code blocks have been removed in https://github.com/numberscope/frontscope/pull/368/commits/b82496784f25066bf84d4361d4bd8d3e2af3aabc. Resolving.